### ISSUE TITLE
proposal: <podcast:images> for multiple image sets

### ISSUE THREAD

**Issue by daveajones** [Podcastindex-org/podcast-namespace#43]
I don't like current tags at all.  They feel wrong.  Clearly there is a need for alternate image sizing.  But, I'm thinking we make it operate more like "alternateEnclosure" and bring in the concept of a srcset like html images:

```
<podcast:alternateImage 
  srcset="https://podcastindex.org/images/pci_avatar.jpg 1500w,
          https://podcastindex.org/images/pci_avatar.jpg 600w,
          https://podcastindex.org/images/pci_avatar.jpg 300w,
          https://podcastindex.org/images/pci_avatar.jpg 150w" />
```

That reduces to a single tag and is very compact.  Would this be workable?

**Comment by snookfin** [comment]
Yes! And there is already a precedent for this type of thing from the itunes namespace.

```
<itunes:owner>
    <itunes:name>Buzzsprout</itunes:name>
    <itunes:email>support@buzzsprout.com</itunes:email>
</itunes:owner>
```

**Comment by MartinMouritzen** [comment]
Better than the existing proposal. I like it.

The only concern I would have is if enough people understand srcset's, and for that reason alone I might be more prone to simply go with something like (off the top of my head):

```<podcast:alternateImage src="x.jpg" width="400" />
<podcast:alternateImage src="x.jpg" width="800" />
<podcast:alternateImage src="x.jpg" width="1200" />
<podcast:alternateImage src="x.jpg" width="1600" />```

(not sure if `"width"` is the right attribute name though) but it depends on who our target audience is. If it's the platforms and "techy" people, I don't think it matters that much, but if we want the semi-techy people with us, I think single tags might be easier to understand and parse.

**Comment by snookfin** [comment]
The srcset code is nice, but I agree it's a little tougher on the eyes. I could go either way.

Here's my question. Why `<podcast:alternateImage>` vs. `<podcast:image>`?

I would think we could just use :image and then you can pass in one or multiple.

**Comment by cisene** [comment]
This alignes with #40 .. secret .. would it be very broken to think about it as that only one URI would be needed and it's companion "documents" (pre-scaled versions of image artwork) would be local and relative to that -- thinking it would reduce length of the attribute some.

**Comment by vandys** [comment]
Since srcset _directly_ maps into how I'd want to use an image, it seems fine to me.  With any other format, I'll just be writing the code to turn it into a srcset.  Just be sure the default src= value is available, and then srcset for the rest.

**Comment by daveajones** [comment]
> 
> 
> The srcset code is nice, but I agree it's a little tougher on the eyes. I could go either way.
> 
> Here's my question. Why `<podcast:alternateImage>` vs. `<podcast:image>`?
> 
> I would think we could just use :image and then you can pass in one or multiple.

I agree with this.  <alternateEnclosure> makes sense, seeing as how the <enclosure> element itself is so foundational.  A different name seems appropriate.  And it describes what it's trying to accomplish.  <podcast:image> doesn't seem to hurt anything though.  Let's go with that.  I'll rewrite it once we hash out the other part.

**Comment by daveajones** [comment]
> 
> 
> Better than the existing proposal. I like it.
> 
> The only concern I would have is if enough people understand srcset's, and for that reason alone I might be more prone to simply go with something like (off the top of my head):
> 
> `<podcast:alternateImage src="x.jpg" width="150" />`
> `<podcast:alternateImage src="x.jpg" width="300" />`
> `<podcast:alternateImage src="x.jpg" width="600" />`
> `<podcast:alternateImage src="x.jpg" width="1500" />`
> 
> (not sure if `"width"` is the right attribute name though) but it depends on who our target audience is. If it's the platforms and "techy" people, I don't think it matters that much, but if we want the semi-techy people with us, I think single tags might be easier to understand and parse.

I can parse anything in the aggregator.  It's not going to bother us at all.  I was mostly thinking that for app developers it would be nice to make the srcset thing work, and you could map that straight into your code.  But, I'll let you be the judge of that.  You are parsing the feeds directly in your app, right Martin?

One thing we're trying to do is reduce barriers to adoption for hosts and developers both.  So a balance between not too many tags (so the feed is bloated), yet not so compact that it's hard to parse (in a perhaps limited power web or mobile app) is important.

**Comment by jamescridland** [comment]
Do you mean:

```
<podcast:alternateImage 
  srcset="https://podcastindex.org/images/pci_avatar-massive.jpg 1500w,
          https://podcastindex.org/images/pci_avatar-middle.jpg 600w,
          https://podcastindex.org/images/pci_avatar-small.jpg 300w,
          https://podcastindex.org/images/pci_avatar-tiny.jpg 150w" />
```

...i.e. the image URLs are actually different?

**Comment by daveajones** [comment]
Yes, that's right James.  Packing them into a single element like that seems more efficient, like img srcset works.

I like `<podcast:images>` a LOT.

**Comment by theDanielJLewis** [comment]
I really like adopting the srcset image standard syntax.

**Comment by daveajones** [comment]
Let me make that change then and then we can let it simmer for a bit.

**Comment by daveajones** [comment]
I made this change and the tag is in the PC20 feed now for testing:   http://mp3s.nashownotes.com/pc20rss.xml

**Comment by daveajones** [comment]
I think it's fair to say that nobody wants 3 image tags. :-)  So, the only real discussion is just whether or not "srcset" works.  

Does anyone thing it _wont_ work?

If not, I'd say it's about as efficient/compact as it can get, and maybe we move on?

**Comment by KieranMcKeefery** [comment]
110% think srcset is the best way to go here, works great for simply just using the value of podcast:images straight into any img tags.

**Comment by daveajones** [comment]
Let's close this issue out then.  If nothing else comes up with `<podcast:images>` we'll move ahead with it.

**Comment by Tombarr** [comment]
My preference is `<podcast:images>` and to stick with the `srcset` attribute as this already has an existing standard and can work drop-in for web-based applications.

Having both thumbnails & large images is of benefit to podcast clients, that no longer need to generate their own thumbnails or hot-link to Apple Podcast thumbnails, and to hosts & listeners who benefit from reduced bandwidth (i.e. not downloading a 3000x3000px image just to scale down to 128px).

**Comment by daveajones** [comment]
Agreed.  Changed the title of the issue to clear up confusion.
